src/lib/protocols/connections/ConnectionService.ts

```diff
@@ -128,18 +129,15 @@ class ConnectionService extends EventEmitter {
     return createOutboundMessage(connectionRecord, response);
   }
 
-  async acceptAck(inboundMessage: InboundMessage<AckMessage>) {
-    const { recipient_verkey, sender_verkey } = inboundMessage;
-    const connectionRecord = await this.findByVerkey(recipient_verkey);
+  async acceptAck(messageContext: MessageContext<AckMessage>) {
+    const connection = messageContext.connection;
 
-    if (!connectionRecord) {
-      throw new Error(`Connection for verkey ${recipient_verkey} not found!`);
+    if (!connection) {
+      throw new Error(`Connection for ${messageContext.recipientVerkey} not found!`);
     }
 
-    validateSenderKey(connectionRecord, sender_verkey);
```

Comment:
The LGTM warning is from the `validateSenderKey` that isn't used anymore. I now do this check at the start after a message is received.

I didn't remove the function however because it contains comments with questions that may still need to be resolved?

Reply:
You can remove it. I still have these questions unanswered, but we the problem has been moved to `MessageReceiver` to line with `if (connection && connection.theirKey != null && connection.theirKey != senderKey) {`. We can discuss it and eventually write some comment there.

Reply:
Allright, removed in next commit.

I added it to the agenda for tomorrow